modules/candidate_parameters/ajax/formHandler.php

```diff
@@ -149,6 +149,44 @@ function editProbandInfoFields($db, $user)
                     ];
 
     $db->update('candidate', $updateValues, ['CandID' => $candID]);
+    foreach (array_keys($_POST) as $field) {
+        if (!empty($_POST[$field])) {
+            if (substr($field, 0, 4) === 'PTID') {
+                $ptid = substr($field, 4);
+
+                $updateValues = [
+                                 'ParameterTypeID' => $ptid,
+                                 'CandID'          => $candID,
+                                 'Value'           => $_POST[$field],
```

Comment:
I'm not very familiar with this module but $_POST[$field] should be sanitized here using [`htmlentities`](https://secure.php.net/manual/en/function.htmlentities.php) unless you know it's explicitly treated in a safe way in the front-end.

Reply:
Actually it should be done on all of the POST variables here...

Reply:
Even if sanitized in the front end by JS, you have to sanitized on the server side as a POST request can be forged.

Reply:
That's not what I meant exactly. I think what you're speaking of is technically _validation_ (sorry to be pedantic) and yes we need to do that both in the front- and back-ends.

Sanitization means treating something safely when it crosses the boundary of one layer of software to another. What I mean here is that someone could enter valid HTML in a POST request (forged or otherwise) which would be saved by the database. When that data leaves the database and is injected into the front-end, we must make sure that it is printed safely (ie. as text) rather than unsafely (ie. as mark up code that executes).